implants/lib/eldritch/src/assets/read_binary_impl.rs

```diff
@@ -41,4 +86,91 @@ mod tests {
         );
         Ok(())
     }
+
+    pub fn init_logging() {
+        pretty_env_logger::formatted_timed_builder()
+            .filter_level(log::LevelFilter::Info)
+            .parse_env("IMIX_LOG")
+            .init();
+    }
+
+    #[tokio::test]
+    async fn test_asset_read_binary_remote() -> anyhow::Result<()> {
+        init_logging();
+        // Create files
+        let tc = Tome {
+            eldritch: r#"print(assets.read_binary("remote_asset/just_a_remote_asset.txt"))"#
+                .to_owned(),
+            parameters: HashMap::new(),
+            file_names: Vec::from(["remote_asset/just_a_remote_asset.txt".to_string()]),
+        };
+
+        // Run Eldritch (in it's own thread)
+        let mut runtime = crate::start(123, tc.clone()).await;
+
+        // We now mock the agent, looping until eldritch requests a file
+        // We omit the sleep performed by the agent, just to save test time
+        loop {
+            // The runtime only returns the data that is currently available
+            // So this may return an empty vec if our eldritch tokio task has not yet been scheduled
+            let messages = runtime.collect();
+            let mut fetch_asset_msgs: Vec<&FetchAssetMessage> = messages
+                .iter()
+                .filter_map(|m| match m {
+                    Message::FetchAsset(msg) => Some(msg),
+                    _ => None,
+                })
+                .collect();
+
+            // If no asset request is yet available, just continue looping
+            if fetch_asset_msgs.is_empty() {
+                continue;
+            }
+
+            // Ensure the right asset was requested
+            assert!(fetch_asset_msgs.len() == 1);
+            let msg = fetch_asset_msgs.pop().expect("no asset request received!");
+            assert!(msg.name == "remote_asset/just_a_remote_asset.txt");
+
+            // Now, we provide the file to eldritch (as a series of chunks)
+            msg.tx
```

Comment:
eventually we will setup mock testing to make this even nicer but this looks nice for now!